### ISSUE TITLE
Drag & Drop not working as expected in all-in-one Touch Screen (issue with touch/mouse events)

### ISSUE THREAD

**Issue by sural98** [gantry/gantry5#39]
I can neither resize a particular particle or position or drag-and-drop a new particle.
Does drag-and-drop feature already work in Layout mode of beta 1?
If yes what am I missing in Opera 29.0.1795.30?

**Comment by sural98** [comment]
Confirmed it is not working with Opera 29.0.1795.30.

**Comment by w00fz** [comment]
Well I tried both 28 stable and 30 developer (I couldn't find 29, I think it's been already replaced by 30) and it works fine for me.
I can drag and drop like in all other browsers. Are you sure you are up to date with it?

**Comment by rhukster** [comment]
Confirmed I was also able to drag and drop and reorder no problem with version 28 stable.

**Comment by w00fz** [comment]
Chrome is my main browser that I use for development and I would have definitely noticed if the Layout Manager drag-n-drop wasn't working.
I haven't heard from anyone else having issues with this, we tested windows, linux, mac and all browsers including tablets and phone.

Perhaps you could give more details regarding the OS you are running and the setup you have. Also if you open the inspector, are you getting any JS error?

Thanks!

**Comment by sural98** [comment]
./assets/js/main.js  of lib_gantry5 package (line 2152) and
./admin/js/main.js of com_gantry5 package (line 18679) have this line:
//# sourceMappingURL=main.js.map
Opera and Chrome dev tools say they can't find "main.js.map"

**Comment by w00fz** [comment]
That shouldn't matter with the drag and drop, it's just the source maps, if it doesn't find them it will continue working as it should. I was hoping for some other js error but there's nothing. What os are your on? 

// Djamil Legato // From iPhone //

**Comment by sural98** [comment]
It is 64-bit Windows 7, WAMP server 2.5

**Comment by sural98** [comment]
I have replicated this issue on a fresh Joomla 3.3.6 of no sample content also

**Comment by JoomFX** [comment]
Works just fine for me on Win7 64bit with both Chrome 41.0.2272.101 and 41.0.2272.118

**Comment by gyvermac** [comment]
No problem on MAC OS 10.10.2  with Safari 8.0.4 or Firefox 36.0.1

**Comment by JoomFX** [comment]
Also tested it with Win 7 64bit and Opera 27.0 and 28.0.1750.48 - everything works just fine.

**Comment by sural98** [comment]
![1920x1080-safari](https://cloud.githubusercontent.com/assets/1849490/6983544/e5a94f02-da27-11e4-8678-ad9556cdf74b.JPG)
This is the screen-shot on Safari with 1920x1080 res. It needs styling.
I can drag-and-drop/resize position, particle however cursor icon won't change its shape "arrows" to that of resizing.
Can you also notice the horizontal widths of 76/24 module positions in the following shot of a different browser size? Horizontal width won't reflect the actual percentage.
![pct-of-positions](https://cloud.githubusercontent.com/assets/1849490/6983719/f54bec88-da29-11e4-9b95-360cd8d39658.jpg)
In short, I have no idea why drag-drop-resize won't work in Chrome, Opera but function in FF and sort of Safari.

**Comment by w00fz** [comment]
You shouldn't use Safari in Windows since it's been discontinued and never updated since v5.1.7 (on Mac latest version is v8.0.4). You can't even download anymore from apple website.

As of why you are having issues with Chrome and Opera I am not sure and I cannot reproduce. Everyone I asked to try with your same setup can drag and resize properly. Perhaps you have another or a friends computer you can try with and see if perhaps is something with your machine configuration?

**Comment by 810** [comment]
Working on Windows 10
- Spartan +1
- Ie 11 +1
- Chrome 41.0.2272.118 +1
- FF 38 DEV

**Comment by sural98** [comment]
I have moved to XAMPP from WAMP if it was WAMP-related.
I run G5+Hydrogen on a fresh J 3.4.1 of no content.
It is still persisting on Chrome and Opera while running smoothly with Firefox.
IE 11.0.9600 can't even load the presets. Layout comes as empty.
Can you suggest some options so that I can debug?

**Comment by w00fz** [comment]
Is there anyway you can expose your site to public so that I can access it via browser and perhaps understand if the issue is server or client related?
If not, do you have another computer you can try reaching the site with and see if perhaps it's your machine setup having issues?

**Comment by sural98** [comment]
Can we arrange a Teamviewer session?

**Comment by w00fz** [comment]
Why don't you hop on Gitter chat on Monday so we can easily talk back and forth and perhaps arrange the teamviewer session? Github comments is definitely making this harder than a proper chat would do :)

[![Join the chat at https://gitter.im/gantry/gantry5](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/gantry/gantry5?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

**Comment by sural98** [comment]
I am now on Gitter's private chat session and waiting for your response. Thanks.

**Comment by w00fz** [comment]
Alright, @sural98 was nice enough to put some light on the problem and him and I just had a TeamViewer session where I could see what was going on. This is a legit issue caused by the JS trying to detect whether it should use `touch` events vs `mouse` events. It is also an issue only happening on all-in-one touch screen when trying to drag & drop and where you can use both a mouse or your finger to control the OS UI. It affects Layout Manager and Menu Edito, probably also sorting the collections / key value fields is not working.

Because both the events are supported and because the JS only implements the first-found event, the default behavior in G5 is to stop at touch events if found ignoring the mouse events.

This is a tricky issue because on all-in-one touch screen you can jump between mouse and finger anytime, it's not fixed.

I have to do some research on this and apparently it is an issue that [other have discussed](http://www.stucox.com/blog/you-cant-detect-a-touchscreen/) already as well.

At the moment the solution provided by @Aesine in https://github.com/gantry/gantry5/issues/61#issuecomment-90106754 is a good workaround if you are using Chrome or Opera.

1. Open chrome://flags/
2. Navigate to `Enable touch events Mac, Windows, Linux, Chrome OS`
3. Disable

Thank you for the time dedicated tracking this issue, hopefully I'll find a way to get around this.

**Comment by w00fz** [comment]
Hey @sural98 and @Aesine I have a custom package that might fix this issue and would like for you guys to try it. If you can hop on Gitter I can send it to you.

**Comment by w00fz** [comment]
Thank you for the help tracking down this issue! I know merged the feature fix into our develop branch and will make it in Beta 3!
